pkg/occlient/occlient.go

```diff
@@ -1306,7 +1313,59 @@ func (c *Client) DeleteProject(name string) error {
 	if err != nil {
 		return errors.Wrap(err, "unable to delete project")
 	}
-	return nil
+
+	// wait for delete to complete
+	w, err := c.projectClient.Projects().Watch(metav1.ListOptions{
+		FieldSelector: fields.Set{"metadata.name": name}.AsSelector().String(),
+	})
+	if err != nil {
+		return errors.Wrapf(err, "unable to watch project")
+	}
+
+	defer w.Stop()
+	for {
```

Comment:
Can we remove this wait for delete confirmation?

Reply:
No, I think that would possibly lead the user into a state of confusion...
Because, the delete takes some time for the project to vanish from the project list and the delay is noticeable.. Now, if we return from delete quickly as soon as we issue the delete api call, and if for some reason the user lists the projects and sees the deleted project still in the list, that would be misleading...

And please note, that the delete of the project involves deletion of all/any resources inside the project which is expected in my opinion to take some time immaterial of whether or not there are any resources within the project(atleast the lookup for presence of any resource has to happen internally..) And whenever the, delete is optimised within the openshift , this watch will also return quickly... optimizing things on our side as well...

WDYT? @syamgk

Reply:
@syamgk Please check https://github.com/redhat-developer/odo/pull/740#issuecomment-427768206 which necessiates having to wait until the project is completely deleted...

The only other alternative to not waiting for complete delete, is to not display current project which was the state of the PR earlier but changed in accordance with the comments in order to maintain consistency with the behaviour of delete of other resources...

I feel correctness and completeness is more important than quickness(specially if its a inherent quality of the underlying components).. So, consider a command like component push which is a considerably heavy-weight operation which takes its due time to complete ... But I am open to any suggestions and will not mind changing the PR accordingly... 

@kadel @cdrage @surajnarwade WDYT?